OpenTrons/opentrons-api
# fix(app): close maintenance run modal if run gets deleted

Fix RQA-1061



# Overview


When a maintenance procedure is canceled from the "Remote Activity" takeover modal on ODD, the corresponding modal on the desktop app should close.

# Test Plan

1. Start a pipette or gripper flow from a desktop app
2. Cancel remote activity from the ODD takeover modal
3. See the modal on desktop close within 5 seconds



# Changelog
1. Refactor chainMaintenanceRunCommands to take the runId in the callback instead of on instantiation
2. Update Gripper flows, Pipette flows, and LPC to pass the runId in the callback
3. Add a `useRef` to keep track of the previous maintenance run id - if this was previously a string and becomes null, we close the gripper and pipette flows since this means the run has been deleted



# Review requests
Run through the test plan, look over approach


# Risk assessment
Low
